hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/h2/H2UpsertQueryBuilderTest.java

```diff
@@ -18,38 +18,33 @@
 
 import com.hazelcast.jet.sql.impl.connector.jdbc.JdbcTable;
 import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.dialect.CalciteSqlDialect;
 import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
-import org.mockito.invocation.InvocationOnMock;
 
 import java.util.Arrays;
+import java.util.Collections;
 
 import static org.junit.Assert.assertEquals;
-import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.when;
 
 public class H2UpsertQueryBuilderTest {
     @Mock
     JdbcTable jdbcTable;
 
-    @Mock
-    SqlDialect sqlDialect;
+    SqlDialect sqlDialect = CalciteSqlDialect.DEFAULT;
```

Comment:
we are mixing dialects here. With `@Mock` it was pure unit test

Reply:
What's my action point? :-)

`JdbcTable` has dialect field, mocking it was getting unwieldy. I consider mock returning a mock a code smell, this seems straightforward. Do you want to do anything with it?

Or are you pointing out that it should use `H2SqlDialect`?

Reply:
> mocking it was getting unwieldy.
That is a sign that the `JdbcTable` or `H2UpsertQueryBuilder` may be not testable easily. But using non-trival real class in unit test is also not beautiful.

Using `H2SqlDialect` would make the code at least consistent so I'd prefer it over `CalciteSqlDialect.DEFAULT`.

Reply:
Maybe `H2UpsertQueryBuilder` should have dialect hardcoded and check if if got table for this dialect? Maybe tables should not have dialects?

Reply:
3ebb1d32ee3